plugins/inputs/monit/monit.go

```diff
@@ -0,0 +1,393 @@
+		fields["status_code"] = service.Status
+		fields["status"] = serviceStatus(service)
+		fields["monitoring_status_code"] = service.MonitoringStatus
+		fields["monitoring_status"] = monitoringStatus(service)
+		fields["monitoring_mode_status"] = monitoringMode(service)
+		fields["monitoring_mode_code"] = service.MonitorMode
+		tags["service"] = service.Name
+		tags["service_type"] = service.Type
+		if service.Type == "0" {
+			fields["mode"] = service.Mode
+			fields["block_percent"] = service.Block.Percent
+			fields["block_usage"] = service.Block.Usage
+			fields["block_total"] = service.Block.Total
+			fields["inode_percent"] = service.Inode.Percent
+			fields["inode_usage"] = service.Inode.Usage
+			fields["inode_total"] = service.Inode.Total
+			acc.AddFields("filesystem", fields, tags)
+		} else if service.Type == "1" {
+			fields["permissions"] = service.Mode
+			acc.AddFields("directory", fields, tags)
+		} else if service.Type == "2" {
+			fields["size"] = service.Size
+			fields["permissions"] = service.Mode
+			acc.AddFields("file", fields, tags)
+		} else if service.Type == "3" {
+			fields["cpu_percent"] = service.CPU.Percent
+			fields["cpu_percent_total"] = service.CPU.PercentTotal
+			fields["mem_kb"] = service.Memory.Kilobyte
+			fields["mem_kb_total"] = service.Memory.KilobyteTotal
+			fields["mem_percent"] = service.Memory.Percent
+			fields["mem_percent_total"] = service.Memory.PercentTotal
+			fields["service_uptime"] = service.Uptime
+			fields["pid"] = service.Pid
+			fields["parent_pid"] = service.ParentPid
+			fields["threads"] = service.Threads
+			fields["children"] = service.Children
+			acc.AddFields("process", fields, tags)
+		} else if service.Type == "4" {
+			fields["hostname"] = service.Port.Hostname
+			fields["port_number"] = service.Port.PortNumber
+			fields["request"] = service.Port.Request
+			fields["response_time"] = service.Port.ResponseTime
+			fields["protocol"] = service.Port.Protocol
+			fields["type"] = service.Port.Type
+			acc.AddFields("remote_host", fields, tags)
+		} else if service.Type == "5" {
+			fields["cpu_system"] = service.System.CPU.System
+			fields["cpu_user"] = service.System.CPU.User
+			fields["cpu_wait"] = service.System.CPU.Wait
+			fields["cpu_load_avg_1m"] = service.System.Load.Avg01
+			fields["cpu_load_avg_5m"] = service.System.Load.Avg05
+			fields["cpu_load_avg_15m"] = service.System.Load.Avg15
+			fields["mem_kb"] = service.System.Memory.Kilobyte
+			fields["mem_percent"] = service.System.Memory.Percent
+			fields["swap_kb"] = service.System.Swap.Kilobyte
+			fields["swap_percent"] = service.System.Swap.Percent
+			acc.AddFields("system", fields, tags)
+		} else if service.Type == "6" {
+			fields["permissions"] = service.Mode
+			acc.AddFields("fifo", fields, tags)
+		} else if service.Type == "7" {
+			fields["last_started_time"] = service.Program.Started * 1000
+			fields["program_status"] = service.Program.Status
+			fields["output"] = service.Program.Output
```

Comment:
Is this the program output?  If so we should probably skip this item due to potential length.  If we would like to keep it then we need to convert/verify that the output is UTF-8 encoded and that the output isn't too large.

Reply:
Done. Please check